xbmc/addons/AddonManager.cpp

```diff
@@ -220,9 +220,13 @@ VECADDONS CAddonMgr::GetAvailableUpdates() const
   m_database.GetRepositoryContent(all_addons);
   std::map<std::string, AddonPtr> latest_private_versions;
   std::map<std::string, AddonPtr> latest_trusted_versions;
+
+  const int prefUpdateMode = CAddonSystemSettings::GetInstance().GetAddonPreferredUpdateMode();
+  CLog::Log(LOGDEBUG,"ADDONS: preferred update mode set to : {}", prefUpdateMode);
+
   for (const auto& addon : all_addons)
   {
-    if (IsFromTrustedRepo(addon))
+    if (prefUpdateMode == UPDATEMODE_ANY_REPOSITORY || IsFromTrustedRepo(addon))
```

Comment:
If an enum class is used then this becomes:

`prefUpdateMode == AddonTrustedUpdateMode::ANY_REPOSITORY`